### PR TITLE
Ga4 accordion link tracking

### PR DESCRIPTION
**Pull request by JamesCGDS**
## What
This PR adds GA4 tracking to accordion links. 

This PR also includes a change to the `setIndex` function and it will now combine `data-ga4-index` (our method of adding `index_link` to links that are generated by content publishers ([further details here)](https://github.com/alphagov/govuk_publishing_components/pull/3262)) with the `index` property if it exists on a parent element.

## Why
Part of the GA4 migration.

The change to `setIndex` has been made because in some scenarios, such as accordion links, we need to set `data-ga4-link` on a parent element due to the links being generated by a content publisher. Within `data-ga4-link`, an `index` property may already exist and contain values such as `index_section` and `index_section_count`. In order to prevent this data from being overwritten with the value in `data-ga4-index` (the previous implementation set `index` to the value of `data-ga4-index` if it existed), `setIndex` combines them together.

[Trello card](https://trello.com/c/F8LlOd07/490-add-tracking-accordion-links)

## Visual Changes
N/A

### CODE REVIEW

File: app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js
Comment (AshGDS) [inline app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js:119]: Could this be done when the indexes are first set (the `setIndexes` function) instead? Just because we set create `data-ga4-index` ourselves, but then it seems like we're fixing it here, when we have control over both functions. Or would doing it in there instead affect other components?

File: app/views/govuk_publishing_components/components/_accordion.html.erb
Comment (AshGDS) [inline app/views/govuk_publishing_components/components/_accordion.html.erb:54]: I think you could get away with not having this comment, I think it's clear from the code that it's separate from the other data_attributes.

File: spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.spec.js
Comment (AshGDS) [inline spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.spec.js:1]: I guess this is something for a separate PR but as we spoke about we need to make sure we always push `index_link: undefined, index_section: undefined, index_section_count: undefined` etc. when only one or two of the index keys have values, as I believe GA4 persists values from previous pushes so we need to make sure empty values are still pushed with their value set to `undefined`. (It's fine to just do `index: undefined` when none of the index keys exist though)

**Review by AshGDS — COMMENTED** [review]
Looking good, had some comments which are more questions than requested changes.

File: app/views/govuk_publishing_components/components/_accordion.html.erb
Comment (JamesCGDS) [reply app/views/govuk_publishing_components/components/_accordion.html.erb:54]: Cool, I'll remove it

File: spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.spec.js
Comment (JamesCGDS) [reply spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.spec.js:1]: I think this might be for another PR - I did discuss this with Andy and we're not sure if `index_section` etc would persist because we'd be overwriting the `index` object itself with a new `index` object if that makes sense. Although we're not 100% sure so I'm going to pick this up with the PA's next week.

File: app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js
Comment (JamesCGDS) [reply app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js:119]: Potentially yeah, I'll need to have a quick think about it

**Review by andysellick — COMMENTED** [review]
Good improvements, nice job, that function is a lot more understandable and simpler now. I'm happy with this but I think Ash has some comments outstanding, so I'll leave him to approve it when he's happy.

**Review by AshGDS — APPROVED** [review]
All good :+1:

**Comment by JamesCGDS** [comment]
Thanks @AshGDS @andysellick - I'll squash the commits together but I'm going to hold off on merging this as the PA's have asked for this not to be deployed yet.

**Pull request merged**
